tests/test_manager.py

```diff
@@ -75,7 +110,25 @@ def get_path_nosync(fid_manager, id):
     if path is None:
         return None
 
-    return os.path.relpath(path, fid_manager.root_dir)
+    return os.path.relpath(path, os.path.normcase(fid_manager.root_dir))
+
+
+def normalize_path(fid_manager: BaseFileIdManager, path: str) -> str:
+    """Normalize path or case based on operating system and FileIdManager instance.
+
+    When testing instances of LocalFileIdManager, we need to normalize the
+    case relative to the OS when comparing results of get_path() since Windows
+    is case-insensitive.
+
+    When testing instances of ArbitraryFileIdManager, we need to normalize the
+    path, regardless of OS, when comparing results of get_path() since this fileID
+    manager must be filesystem agnostic.
+    """
+    if isinstance(fid_manager, LocalFileIdManager):
+        path = os.path.normcase(path)
+
+    parts = path.strip("\\").split("\\")
+    return "/".join(parts)
```

Comment:
I don't quite understand why this is necessary. Here me out:

1. The paths we use in the fixtures are always lower case and relative. So `normcase` shouldn't be doing anything.
2. We're now always using `posixpath.join()` in the fixtures, which uses `/` as the separator. So lines 130-131 shouldn't be doing anything either.

I think the changes you made to the fixtures are sufficient. Can we try removing this helper and see if CI still passes for Windows?

Reply:
> 1. The paths we use in the fixtures are always lower case and relative. So normcase shouldn't be doing anything.

The test "paths" are indeed.  The test `root_dir` values are not.   They are represented by the pytest temporary directories and, on Windows at least, contain some uppercase characters - which `normcase()` lowers on Windows.

> 2. We're now always using posixpath.join() in the fixtures, which uses / as the separator. So lines 130-131 shouldn't be doing anything either.

True, but even `posixpath.join()` won't convert _embedded_ backslashes within one of its arguments.  e.g., `posixpath.join("c:\\path\\to\\temp\\folder", "my/api/path")` yields `c:\\path\\to\\temp\\folder/my/api/path`.

The above yields: `c:/path/to/temp/folder/my/api/path`

Reply:
> The test root_dir values are not. They are represented by the pytest temporary directories and, on Windows at least, contain some uppercase characters - which normcase() lowers on Windows.

Hmm, can we just normalize the root directory argument in the `*fid_manager` fixtures in `pytest_plugin.py` instead then?

I just don't want to force usage of `normalize_path()` whenever comparing paths in the unit tests. This hinders readability and can cause loss of valuable developer time if they forget to call `normalize_path()` before assertion.

> even posixpath.join() won't convert embedded backslashes within one of its arguments. e.g., posixpath.join("c:\\path\\to\\temp\\folder", "my/api/path") yields c:\\path\\to\\temp\\folder/my/api/path.

Right, but this function will always receive a test path, which is guaranteed to be relative and generated via `posixpath.join()`. Can you try removing the backslash => forwards slash logic and see if it still works?

Reply:
> Right, but this function will always receive a test path, which is guaranteed to be relative and generated via posixpath.join(). Can you try removing the backslash => forwards slash logic and see if it still works?

Again, it's the `root_dir` values that are the issue.  When they contain multiple "directories" _within_ a parameter of `join()`, those separators will not be converted.  Your idea of incorporating this in the fixture is reasonable and I'll take a look.

Another approach may be to massage `root_dir` within its validator such that it's always in a form that will be persisted.  

One of the primary problems I ran into is that the tests start with filesystem-path files but need to compare against, normalized versions of those (i.e., the "api path"), and since the test parameters are both FileSystem and NonFileSystem implementations and *nix/Windows, comparing inputs that are solely filesystem-derived requires a conversion.

I'll revisit this though - thanks.